promtool "panic: assignment to entry in nil map"

## Bug Report
**What did you do?**

I just started playing with unittests for alerting rules. Below is the content of rule and test files:

rules_example.yml:
```
groups:

  - name: Decrease
    interval: 1m
    rules:
    - alert: "Decrease"
      expr: ((sum(total_for_each_active_hotel:avg:5m offset 10m) by (chainName,reportId) - sum(total_for_each_active_hotel:avg:5m) by (chainName,reportId)) / (sum(total_for_each_active_hotel:avg:5m) by (chainName,reportId))) * 100 > 15
      annotations:
        summary: 'Decrease'
        description: 'Decrease is over 15%, chain: "{{ $labels.chainName }}" report_id: "{{ $labels.reportId }}"'
        value: '{{ $value }}'
        chain: '{{ $labels.chainName }}'
        report_id: '{{ $labels.reportId }}'
        source: Prometheus
```
test_example.yml:
```
rule_files:
  - ./rules_example.yml

tests:
  - interval: 5m
    input_series:
      - series: total_for_each_active_hotel:avg:5m{chainName="BetterStay",cluster="West",hotelName="Not so nice resort",reportId="1234567"}
        values: '10 10 10 10 10 0 0'
      - series: total_for_each_active_hotel:avg:5m{chainName="BetterStay",cluster="West",hotelName="Nice Resort",reportId="1234567"}
        values: '150 150 150 140 100 100 100'
    alert_rule_test:
      - alertname: "Decrease"
        exp_alerts:
          - exp_labels:
              chainName: BetterStay
              reportId: 1234567
          - exp_annotations:
              description: 'Decrease is over 15%, chain: "BetterStay" report_id: "1234567"'
              summary: 'Decrease'
              chain: BetterStay
              report_id: 1234567
              source: Prometheus

```
It is a 'toy' configuration I just build up to familiarize myself with unit tests. However, when I run 
`promtool test rules test_example.yml` I've got the following output:
```
Unit Testing:  test_example.yml
panic: assignment to entry in nil map

goroutine 1 [running]:
main.(*testGroup).test(0xc000811a28, 0x0, 0xe7791f700, 0x254c500, 0x0, 0xe7791f73c, 0x254c500, 0xdf8475800, 0xc00002f9f8, 0xc00053bcc0, ...)
	/app/cmd/promtool/unittest.go:289 +0x1178
main.ruleUnitTest(0x7ffcaa4d20cb, 0x10, 0xc000146a80, 0x2, 0x2)
	/app/cmd/promtool/unittest.go:101 +0x5d7
main.RulesUnitTest(0xc00053b660, 0x1, 0x1, 0x1)
	/app/cmd/promtool/unittest.go:44 +0x192
main.main()
	/app/cmd/promtool/main.go:147 +0x2250
```

I assume that even if the configuration is bad/broken it should not panic. Looks like a bug.

Prometheus 2.10 and 2.11.1

--- a/cmd/promtool/unittest.go
+++ b/cmd/promtool/unittest.go
@@ -286,6 +286,9 @@ func (tg *testGroup) test(mint, maxt time.Time, evalInterval time.Duration, grou
 				for _, a := range testcase.ExpAlerts {
 					// User gives only the labels from alerting rule, which doesn't
 					// include this label (added by Prometheus during Eval).
+					if a.ExpLabels == nil {
+						a.ExpLabels = make(map[string]string)
+					}
 					a.ExpLabels[labels.AlertName] = testcase.Alertname
 
 					expAlerts = append(expAlerts, labelAndAnnotation{

